src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -497,11 +499,31 @@ export async function sendTelemetryForPythonKernelExecutable(
         }
         const sysExecutable = concatMultilineString(output.text).trim().toLowerCase();
         const match = kernelConnection.interpreter.path.toLowerCase() === sysExecutable;
-        sendTelemetryEvent(Telemetry.PythonKerneExecutableMatches, undefined, {
-            match: match ? 'true' : 'false',
-            kernelConnectionType: kernelConnection.kind
+        if (match) {
```

Comment:
With the previous kernel refactor we seemed to have regressed a in one case where we weren't starting the right interpreter.
That was fixed & will go out in the next release.

However its possible there are other scenarios when this still doesn't work.
Or the telemetry is wrong, i.e. algorithm used to compare interpreters is incorrect.
Improved the code check the paths.